src/mdns/mdns_avahi.cpp

```diff
@@ -517,7 +517,7 @@ otbrError PublisherAvahi::PublishService(uint16_t aPort, const char *aName, cons
         int         rval;
         const char *value       = va_arg(args, const char *);
         size_t      valueLength = va_arg(args, size_t);
-        size_t      needed      = sizeof(AvahiStringList) + strlen(name) + valueLength;
+        size_t      needed      = sizeof(AvahiStringList) + strlen(name) + valueLength + 1;
```

Comment:
nit: add comment to explain the `+ 1` and why the `< sizeof(buffer)` check below (vs. `<= sizeof(buffer)` is not sufficient.

Reply:
Checked the logic and found that the`<= sizeof(buffer)` suffices. Added the comments.